src/compile/selection/interval.ts

```diff
@@ -29,18 +32,43 @@ const interval:SelectionCompiler = {
     }
 
     selCmpt.project.forEach(function(p) {
-      if (p.encoding !== X && p.encoding !== Y) {
+      const channel = p.encoding;
+      if (channel !== X && channel !== Y) {
         warn('Interval selections only support x and y encoding channels.');
         return;
       }
 
-      const cs = channelSignals(model, selCmpt, p.encoding);
-      const csName = channelSignalName(selCmpt, p.encoding, 'data');
+      const cs = channelSignals(model, selCmpt, channel);
+      const dname = channelSignalName(selCmpt, channel, 'data');
+      const vname = channelSignalName(selCmpt, channel, 'visual');
+      const scaleStr = stringValue(model.scaleName(channel));
+
       signals.push.apply(signals, cs);
-      intervals.push(`{encoding: ${stringValue(p.encoding)}, ` +
-      `field: ${stringValue(p.field)}, extent: ${csName}}`);
+      intervals.push(`{encoding: ${stringValue(channel)}, ` +
+        `field: ${stringValue(p.field)}, extent: ${dname}}`);
+
+      scaleTriggers.push({
+        scaleName: model.scaleName(channel),
+        expr: `(!isArray(${dname}) || ` +
+          `(invert(${scaleStr}, ${vname})[0] === ${dname}[0] && ` +
+            `invert(${scaleStr}, ${vname})[1] === ${dname}[1]))`
+      });
     });
 
+    // Proxy scale reactions to ensure that an infinite loop doesn't occur
+    // when an interval selection filter touches the scale.
+    if (!hasScales) {
+      signals.push({
+        name: name + SCALE_TRIGGER,
+        value: {},
+        on: [{
+          events: scaleTriggers.map((t) => { return {scale: t.scaleName}; }),
```

Comment:
No need to change at all but FYI you can do 

```
scaleTriggers.map(t => ({scale: t.scaleName}))
```

(The trick for returning object is to use a parenthesis to wrap it.)